### PR TITLE
Fix Carthage warning when building by including in workspace

### LINKED ISSUE #25: Should not list Carthage in Framework Search Paths
Currently `FRAMEWORK_SEARCH_PATHS ` is set to `$(PROJECT_DIR)/Carthage/Build/iOS`. This gives a warning when building from source in a consuming workspace:

```
ld: warning: directory not found for option '-F/Users/bejones/Development/project/Carthage/Checkouts/RxKeyboard/Carthage/Build/iOS'
```

The correct setup is to not have this and to include RxSwift in the Workspace and use Xcode's Implicit Dependency Management. That way it knows how to compile both as a part of Carthage and in the workspace for development: https://www.quora.com/How-should-an-OSS-library-be-structured-for-distribution-with-both-Carthage-and-CocoaPods/answer/Justin-Spahr-Summers

### PR DESCRIPTION
**Pull request by brentleyjones**
Fixes #25.

This updates the workspace to build with a submodule for RxSwift instead of assuming that the framework will be built to a specific location.

I also updated the Podfile.lock in the Example directory.

I tested this both building by the command line with `carthage` and by including the project directly into a workspace.

### CODE REVIEW

File: Example/Podfile.lock [hunk 1]
```diff
@@ -2,12 +2,12 @@ PODS:
   - CGFloatLiteral (0.2.0)
   - ManualLayout (1.3.0)
   - ReusableKit (1.1.0)
-  - RxCocoa (3.1.0):
-    - RxSwift (~> 3.1)
-  - RxKeyboard (0.4.0):
-    - RxCocoa (>= 3.0)
-    - RxSwift (>= 3.0)
-  - RxSwift (3.1.0)
+  - RxCocoa (3.5.0):
```
Comment (freak4pc) [inline Example/Podfile.lock]: Any reason to increase all the minimum versions in the Podfile.lock ?

Comment (devxoul) [inline Example/Podfile.lock]: Yeah, I think Podfile.lock is an unrelated change.

Comment (brentleyjones) [inline Example/Podfile.lock]: Because without it the example won't compile. If you just run `pod install` it says it can't meet the requirements of the `3.4.0` that the library requires, while being locked to `3.1.0`.

Comment (brentleyjones) [inline Example/Podfile.lock]: I have it as two commits. I can remove that commit if necessary. It does need updating though :smile:.

Comment (freak4pc) [inline Example/Podfile.lock]: Uhm very weird it would require 3.4 , but even if so let's try keep it to the minimum required? 3.5 isn't a requirement here. So besides that I have no problems with this PR

Comment (devxoul) [inline Example/Podfile.lock]: I just remembered that there was https://github.com/RxSwiftCommunity/RxKeyboard/pull/20. I think it makes sense.

**Comment by devxoul** [comment]
Thanks for your contribution!

```
 🎉  Congrats

 🚀  RxKeyboard (0.5.1) successfully published
 📅  July 6th, 16:05
 🌎  https://cocoapods.org/pods/RxKeyboard
 👍  Tell your friends!
```
